Objective-C/Internal/CBLIndexable+Internal.h

```diff
@@ -0,0 +1,31 @@
+//
+//  CBLIndexable+Internal.h
+//  CouchbaseLite
+//
+//  Copyright (c) 2024 Couchbase, Inc All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+#import "CouchbaseLite/CBLIndexable.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@protocol CBLIndexableInternal <CBLIndexable>
```

Comment:
Don't need to create the protocol for this as the method can be put directly to CBLCollection Internal. 

Also no need to have underscored prefix as this function will not be a public API.